packages/design-system-scripts/gulp/docs/generatePages/savePage.js

```diff
@@ -0,0 +1,98 @@
+const crypto = require('crypto');
+const fs = require('mz/fs');
+const path = require('path');
+const recursive = require('mkdir-recursive');
+
+/**
+ * Creates or updates an HTML file if it is new or has changed. If its contents
+ * are the same, no action is taken on the file.
+ * @param {Object} page
+ * @param {String} docsPath
+ * @returns {Promise}
+ */
+function savePage(page, docsPath) {
+  const pathObj = docsFilePath(page.uri, docsPath);
+  const html = template(page);
+
+  return checkCache(html, pathObj.path).then(changed => {
+    if (changed) {
+      return saveToFile(html, pathObj);
+    }
+
+    return Promise.resolve(false);
+  });
+}
+
+/**
+ * Create the HTML for a page
+ * @param {Object} page
+ * @return {String}
+ */
+function template(page) {
+  return `<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  ${page.head}
+</head>
+<body class="ds-base">
+  ${page.body}
+</body>
+</html>`;
+}
+
+/**
+ * To ensure we're not unnecessarily regenerating each page each time the
+ * generate-pages task is called, we first compare the checksums.
+ * @return {Boolean} Should the file be regenerated?
+ */
+function checkCache(html, path) {
+  return fs
+    .readFile(path, 'utf8')
+    .then(data => {
+      const fileHash = crypto
+        .createHash('md5')
+        .update(data)
+        .digest('hex');
+      const htmlHash = crypto
+        .createHash('md5')
+        .update(html)
+        .digest('hex');
+      return fileHash !== htmlHash;
+    })
+    .catch(() => true); // File doesn't exist
+}
+
+/**
+ * @param {String} uri
+ * @param {String} docsPath
+ * @returns {Object}
+ */
+function docsFilePath(uri, docsPath) {
+  if (uri === 'public') throw Error('Filename can\'t be "public"');
+  const dir = path.resolve(docsPath, uri);
```

Comment:
This is an example of a `path.resolve` call that was changed. The old version used to look like this:
```js
const dir = path.resolve(__dirname, '../../../', docsPath, uri);
```

Reply:
Could you remind me why we dont need `__dirname` or `../../../` again? Is it because `docsPath` is now just `packages/design-system-docs/dist`?

Reply:
It's because where we'll be executing the script (the pwd) vs where the script lives isn't always the same now. In the CMSDS repo, we'll be running it from the root, but the script will live in packages/design-system-scripts. In child design systems, it could be anywhere really. `path.resolve(__dirname)` would give you the directory where that specific module is (aka `/Users/whatever/design-system/packages/design-system-scripts/gulp/docs/generatePages/savePage.js`. We don't really care where that specific module is because the only path that matters is where the file we're referencing is relative to where the script is being executed. If you were to execute the script in your home directory,  `path.resolve('.')` will always return your home directory no matter which module is calling it, which is more helpful.